src/Tribe/Views/V2/Views/Traits/HTML_Cache.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * Provides methods for HTML caching for views
+ *
+ * @since   4.9.11
+ *
+ * @package Tribe\Events\Views\V2\Views
+ */
+
+namespace Tribe\Events\Views\V2\Views\Traits;
+
+use Tribe__Cache_Listener as Cache_Listener;
+use Tribe__Context as Context;
+use Tribe__Date_Utils as Dates;
+use Tribe\Events\Views\V2\View;
+
+/**
+ * Trait HTML_Cache
+ *
+ * @since   5.0.0
+ *
+ * @package Tribe\Events\Views\V2\Views
+ *
+ * @property Context $context The current View context.
+ */
+trait HTML_Cache {
```

Comment:
On this one I think it would ideal if we stick with a class type and inject the class into the constructor.

https://github.com/moderntribe/the-events-calendar/pull/2987/files#diff-828c168861c5b8f4cb8331c20c65d55aR216

So in future we can extract and interface of the Cache so we can replace the Cache on the fly using a memory cache for testing or local dev, or we can replace it with a file cache or any other one that just plugins in and implement the interface from there.

Reply:
That also removes the need to prefix every method with `_html` as the class will be injected and assigned to an instance variable like: 

```
$this->cache->maybe_get_cached_data()
$this->cache->maybe_cache();
```

Reply:
Create a ticket for us to handle this: https://moderntribe.atlassian.net/browse/TEC-3147